designs/tls/install-cert-manager.md

```diff
@@ -0,0 +1,187 @@
+---
+oep-number: OEP 3855
+title: Enable TLS in Mayastor
+authors:
+  - "@Johnaius"
+owners:
+  - "@Johnaius"
+editor: TBD
+creation-date: 02/6/2025
+last-updated: 02/6/2025
+status: implementable
+---
+
+# Enable TLS in Mayastor
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [User Stories](#user-stories)
+    * [Story 1](#story-1)
+    * [Story 2](#story-2)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    * [Discoverable images](#discoverable-images)
+      * [Regex Explanation](#regex-explanation)
+    * [Expose Images](#expose-images)
+    * [Non-Discoverable/Runtime images](#non-discoverableruntime-images)
+  * [Test Plan](#test-plan)
+  * [Risks and Mitigations](#risks-and-mitigations)
+    * [Mitigations](#mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+
+## Summary
+
+This proposal aims to add TLS communication between the components of Mayastor
+
+## Motivation
+
+Security
+
+### Goals
+
+1. Create TLS switch
+
+1. Install Cert-Manager
```

Comment:
I wonder, should install cert-manager be a goal, or should the goal be to use cert-manager?
With this a few questions:
1. Should we support using existing cert-manager?
2. Should we make cert-manager a pre-requisite rather than installing it as a dependency? This would also get around the CRD issue?

Reply:
Let's say someone already installed the cert-manager for their other projects.
Can we make use of it?

Reply:
Ah, found it. We have jaeger tracing for debugging, and for it to work, a pre-req is the cert-manager.
<https://github.com/jaegertracing/helm-charts/tree/main/charts/jaeger-operator#prerequisites>
So we could have cert-manager as a pre-requisite?
This brings other questions, can multiple instances of cert-manager be installed in a cluster?

Reply:
I believe not I remember reading in the [docs](https://cert-manager.io/v1.12-docs/installation/helm/#:~:text=care%20must%20be%20taken%20to%20ensure%20that%20it%20is%20installed%20exactly%20once) that cert-manager should only be install once in cluster

Reply:
> we could enable tls but disable cert-manager in helm chart perhaps?

Yeah seems like we'd have to do that since cert-manager should only be installed once.
We do similar thing for etcd: <https://github.com/openebs/mayastor-extensions/blob/e620241d89d9d92066da4b2e35e593008fe3c27b/chart/values.yaml#L487>

Btw seems like we can get the crd installed through the cert-manager chart using: `cert-manager.crds.enabled` and `cert-manager.crds.keep`